ethcore/src/state/mod.rs

```diff
@@ -537,13 +548,8 @@ impl<B: Backend> State<B> {
 		let is_value_transfer = !incr.is_zero();
 		if is_value_transfer || (cleanup_mode == CleanupMode::ForceCreate && !self.exists(a)?) {
 			self.require(a, false)?.add_balance(incr);
-		} else {
-			match cleanup_mode {
-				CleanupMode::KillEmpty(set) => if !is_value_transfer && self.exists(a)? && !self.exists_and_not_null(a)? {
-					set.insert(a.clone());
-				},
-				_ => {}
-			}
+		} else if cleanup_mode == CleanupMode::KillEmpty && self.exists(a)? {
+			self.require(a, false)?;
```

Comment:
consider introducing a function `fn touch(&self, a: ...) -> ... { self.require(a, false) }` and using that in order to make clear the intent here.
this doesn't make use of `is_value_transfer`; i am skeptical that precise semantics have been preserved.